### PR TITLE
Fix get or create in chat.ps1, datapolicyoperation.ps1, driveItem.ps1

### LINKED ISSUE #488: Data Policy Operation Exception
To avoid the following exception and aid in reproducibility when executing Personal Data Export, consider deleting `RequestInfo.json` from Blob Storage before executing the following request. 
https://github.com/microsoftgraph/msgraph-sdk-raptor/blob/704c1338b0c228bdd4f13dc5fc4d821d3f392509/scripts/TenantSetup/TenantHttpData/dataPolicyOperation/dataPolicyOperation.ps1#L15-L20
```
{
   "error":{
      "code":"StorageLocationAlreadyUsed",
      "message":"Storage destination has already been used for an export. If this is not true, delete 'RequestInfo.json' from the container and try again.",
      "innerError":{
         "date":"2021-10-07T14:49:41",
         "request-id":"eb2fb33e-2e56-4b4d-80e8-57917f30e356",
         "client-request-id":"eb2fb33e-2e56-4b4d-80e8-57917f30e356"
      }
   }
}
```

### LINKED ISSUE #491: NRE(Null Reference Exception) when handling PrintJob
NRE issue in Printer Request when fetching Printer Jobs. 
https://github.com/microsoftgraph/msgraph-sdk-raptor/blob/704c1338b0c228bdd4f13dc5fc4d821d3f392509/scripts/TenantSetup/TenantHttpData/printer/printer.ps1#L40-L45
image

### LINKED ISSUE #493: Check for existing Print related entities before creating.
For example, when executing `print/taskDefinitions`, we need to check whether entity exists before creating. 
Use [List taskDefinitions](https://docs.microsoft.com/en-us/graph/api/print-list-taskdefinitions?view=graph-rest-1.0&tabs=http) to get existing definitions. 
https://github.com/microsoftgraph/msgraph-sdk-raptor/blob/704c1338b0c228bdd4f13dc5fc4d821d3f392509/scripts/TenantSetup/TenantHttpData/printer/printer.ps1#L47-L52

### LINKED ISSUE #494: Create PrintTaskTrigger odata bind issue.
On issue with odata, possibly caused by errors earlier in the script.  https://github.com/microsoftgraph/msgraph-sdk-raptor/blob/704c1338b0c228bdd4f13dc5fc4d821d3f392509/scripts/TenantSetup/TenantHttpData/printer/printer.ps1#L54-L59
```http
POST https://graph.microsoft.com/v1.0/print/printers//taskTriggers 
HTTP/1.1 400 Bad Request 
Date: Fri, 08 Oct 2021 00:40:34 GMT 
Transfer-Encoding: chunked 
Vary: Accept-Encoding 
Strict-Transport-Security: max-age=31536000 
request-id: ae5c1a11-b399-4ce7-a425-81d227b528c3 
client-request-id: ae5c1a11-b399-4ce7-a425-81d227b528c3 
x-ms-ags-diagnostic: {"ServerInfo":{"DataCenter":"North Europe","Slice":"E","Ring":"4","ScaleUnit":"005","RoleInstance":"DU6PEPF00000C14"}} 
Content-Type: application/json 
Content-Encoding: gzip  
{
   "error":{
      "code":"BadRequest",
      "message":"Invalid bind property name definition in request.",
      "innerError":{
         "date":"2021-10-08T00:40:35",
         "request-id":"ae5c1a11-b399-4ce7-a425-81d227b528c3",
         "client-request-id":"ae5c1a11-b399-4ce7-a425-81d227b528c3"
      }
   }
}



```

### PR DESCRIPTION
**Pull request by fey101**
Do a get or create for the scripts so that we only create if object doesn't exist.

For task trigger, there's a known issue where despite live view returning an empty array, the creation phase fails with error message task trigger already exists. Related issue here https://github.com/microsoftgraph/microsoft-graph-docs/issues/14774

Closes #493 , resolves #494, resolves #491, resolves #488

### CODE REVIEW

File: scripts/TenantSetup/TenantHttpData/dataPolicyOperation/dataPolicyOperation.ps1 [hunk 1]
```diff
@@ -9,16 +9,18 @@ $appSettings = Get-AppSettings
...
-    $dataPolicyOperation = Request-DelegatedResource -Uri "users/$($userId)/exportPersonalData" -Method "Post" -Body $dataPolicyOperationContent
-    $dataPolicyOperation.Location
-    $dataPolicyOperation_id = $dataPolicyOperation.Location -split "/" | Select-Object -last 1
+$userId = Get-AdminUserId
+$exportDataRequest = az storage blob exists --container-name exportpersonaldatastorage --name RequestInfo.json --connection-string $appSettings.AzureStorageConnectionString
+$exportDataExists = $exportDataRequest[1].split(":")[1]
+if ($exportDataExists.trim() -eq "true"){
+    az storage blob delete --container-name exportpersonaldatastorage --name RequestInfo.json --connection-string $appSettings.AzureStorageConnectionString
 }
+$dataPolicyOperationContent = Get-RequestData -ChildEntity "dataPolicyOperation"
```
Comment (fey101) [inline scripts/TenantSetup/TenantHttpData/dataPolicyOperation/dataPolicyOperation.ps1:18]: There is no list endpoint for dataPolicyOperation hence we will need to create every time.

**Review by georgend — CHANGES_REQUESTED** [review]

**Comment by georgend** [comment]
Have you run these changes in the context of InitTenant.ps1 script. As part of the wholly integrated tenant script?

File: scripts/TenantSetup/TenantHttpData/dataPolicyOperation/dataPolicyOperation.ps1 [hunk 2]
```diff
@@ -9,16 +9,18 @@ $appSettings = Get-AppSettings
...
-$userId
-
-if($null -eq $dataPolicyOperation) {
-    $dataPolicyOperationContent = Get-RequestData -ChildEntity "dataPolicyOperation"
-    $dataPolicyOperationContent.storageLocation = $appSettings.SASUrl  # Retrieve SASUrl from azure key-Vault
-    $dataPolicyOperation = Request-DelegatedResource -Uri "users/$($userId)/exportPersonalData" -Method "Post" -Body $dataPolicyOperationContent
-    $dataPolicyOperation.Location
-    $dataPolicyOperation_id = $dataPolicyOperation.Location -split "/" | Select-Object -last 1
+$userId = Get-AdminUserId
+$exportDataRequest = az storage blob exists --container-name exportpersonaldatastorage --name RequestInfo.json --connection-string $appSettings.AzureStorageConnectionString
```
Comment (georgend) [inline scripts/TenantSetup/TenantHttpData/dataPolicyOperation/dataPolicyOperation.ps1:13]: Is `AzureStorageConnectionString` already added to our Azure AppConfig?

File: scripts/TenantSetup/TenantHttpData/driveItem/driveItem.ps1 [hunk 3]
```diff
@@ -104,12 +104,17 @@ $workbookCommentReply = Request-DelegatedResource -Uri "me/drive/items/$($driveI
 $workbookCommentReply.id
 $identifiers.driveItem.workbookComment.workbookCommentReply._value = $workbookCommentReply.id
 
-
-$operationData = Get-RequestData -ChildEntity "driveItemWorkbookOperation"
-$operationHeaders = @{"Prefer"="respond-async"}
-$driveItemWorkbookOperation = Request-DelegatedResource -Uri "me/drive/items/$($driveItem.id)/workbook/createSession" -Method "POST" -Headers $operationHeaders -Body $operationData
+$currentOperationId = $identifiers.driveItem.workbookOperation._value
+if ($currentOperationId){
+    driveItemWorkbookOperation = Request-DelegatedResource -Uri "/me/drive/items/$($driveItem.id)/workbook/operations/$($currentOperationId)"
```
[the remark this answers is not in the document]
Comment (fey101) [reply scripts/TenantSetup/TenantHttpData/driveItem/driveItem.ps1:109]: Nice catch. Thanks

File: scripts/TenantSetup/TenantHttpData/dataPolicyOperation/dataPolicyOperation.ps1 (hunk 2 above)
Comment (fey101) [reply scripts/TenantSetup/TenantHttpData/dataPolicyOperation/dataPolicyOperation.ps1]: It is the same value for RaptorStorageConnectionString so I have switched to using the commonly used name in the updated changes. That makes azureStorageConnectionString irrelevant.

**Review by georgend — APPROVED** [review]

**Pull request merged**